addons/a11y/src/index.ts

```diff
@@ -39,20 +39,29 @@ const run = (element: ElementContext, config: Spec, options: RunOptions) => {
             restoreScroll: true,
           } as RunOptions) // cast to RunOptions is necessary because axe types are not up to date
       )
-      .then(report);
+      .then(report)
+      .catch(error => addons.getChannel().emit(EVENTS.ERROR, String(error)));
   });
 };
 
 if (module && module.hot && module.hot.decline) {
   module.hot.decline();
 }
 
+let storedDefaultSetup: Setup | null = null;
+
 export const withA11y = makeDecorator({
   name: 'withA11Y',
   parameterName: PARAM_KEY,
   wrapper: (getStory, context, { parameters }) => {
     if (parameters) {
-      setup = parameters as Setup;
+      if (storedDefaultSetup === null) {
+        storedDefaultSetup = { ...setup };
```

Comment:
Sorry for the question, but I don't really understand the goal. Isn't the issue about parameters isolation ?

AFAIU, here, when you add parameters, they are stored and become the default parameters.
But next time, when you don't pass any parameters, you use the default one, and reset the this default. Doesn't your previous parameters impact your current stories ?

And everytime, setup is changed, so on EVENTS.REQUEST, the arrow function uses the same setup ref, which content changes.

Am I missing something ?

Reply:
We originally create `setup` as `let setup: Setup = { element: null, config: {}, options: {} };`. I'm not sure when, but at some point it will be re-configured with some defaults (e.g. for `element` and `options` gets some reporter config, if I remember correctly). ⚠️

However as soon as there are `parameters` in the old code `setup` was re-assigned (`setup = parameters as Setup;`) and from that point in time only _that_ setup was used and never reset if you switched to a story _without_ `parameters`.

Apparently some of the code needed to have a _reference_ to the original `setup`.

Because I don't know when the original `setup` ⚠️ was reconfigured, I try to store a shallow copy with the correct config in `storedDefaultSetup` which gets restored as soon as I switch to a story without `parameters`.

That is the basic logic.

It could be better/simplified if I could properly get/create the setup differently. I'd assume we should have
1. some default/base setup,
2. merge the global-level custom config (`import { addParameters } from '@storybook/react'; addParameters({ a11y })`),
3. merge the component-level custom config (`export default { parameters: { a11y } }` _or_ `storiesOf('...', module).addParameters({ a11y })`),
4. merge the story-level custom config (`Example.story = { parameters: { a11y } };` _or_ `storiesOf('...', module).add('...', () => {}, { a11y })`),
and do that every time before `run` is called.

But I don't know how to get 1., 2. and 3. properly (the workaround with `storedDefaultSetup` is currently for these three steps). I only know how to get 4. (which is `parameters`), if I'm not misunderstanding something.

Reply:
You're right, it doesn't seem to keep the origin of each parameters. (`client_api.ts > addParameters()`, it merges the `_globalParameters` directly).

Reply:
You're right, it doesn't seem to keep the origin of each parameters. (`client_api.ts > addParameters()`, it merges the `_globalParameters` directly).

Reply:
Should I create a seperate issue for that and we keep my workaround for now? I can also add a comment where I add a link to the new issue.

Reply:
I think that would be great :)

Reply:
Follow-up story: https://github.com/storybookjs/storybook/issues/8832